src/coreml.cs

```diff
@@ -0,0 +1,273 @@
+		Generic = 0,
+		FeatureType = 1,
+		DescriptionMismatch = 2,
+		Io = 3
+	}
+
+	[Watch (4,0), TV (11,0), Mac (10,13, onlyOn64: true), iOS (11,0)]
+	[Native]
+	public enum MLMultiArrayDataType : nint {
+		Double = 65536 | 64,
+		Float32 = 65536 | 32,
+		Int32 = 131072 | 32
+	}
+
+	[Watch (4,0), TV (11,0), Mac (10,13, onlyOn64: true), iOS (11,0)]
+	[BaseType (typeof (NSObject))]
+	interface MLDictionaryFeatureProvider : MLFeatureProvider {
+
+		[Export ("dictionary")]
+		NSDictionary<NSString, MLFeatureValue> Dictionary { get; }
+
+		[Export ("initWithDictionary:error:")]
+		IntPtr Constructor (NSDictionary<NSString, NSObject> dictionary, out NSError error);
+
+		[Export ("objectForKeyedSubscript:")]
+		[return: NullAllowed]
+		MLFeatureValue ObjectForKeyedSubscript (string featureName);
+	}
+
+	[Watch (4,0), TV (11,0), Mac (10,13, onlyOn64: true), iOS (11,0)]
+	[BaseType (typeof (NSObject))]
+	interface MLFeatureDescription : NSCopying {
+
+		[Export ("name")]
+		string Name { get; }
+
+		[Export ("type")]
+		MLFeatureType Type { get; }
+
+		[Export ("optional")]
+		bool Optional { [Bind ("isOptional")] get; }
+
+		[Export ("isAllowedValue:")]
+		bool IsAllowedValue (MLFeatureValue value);
+	}
+
+	interface IMLFeatureProvider { }
+
+	[Watch (4,0), TV (11,0), Mac (10,13, onlyOn64: true), iOS (11,0)]
+	[Protocol]
+	interface MLFeatureProvider {
+
+		[Abstract]
+		[Export ("featureNames")]
+		NSSet<NSString> FeatureNames { get; }
+
+		[Abstract]
+		[Export ("featureValueForName:")]
+		[return: NullAllowed]
+		MLFeatureValue GetFeatureValue (string featureName);
+	}
+
+	[Watch (4,0), TV (11,0), Mac (10,13, onlyOn64: true), iOS (11,0)]
+	[BaseType (typeof (NSObject))]
+	interface MLFeatureValue : NSCopying {
+
+		[Export ("type")]
+		MLFeatureType Type { get; }
+
+		[Export ("undefined")]
+		bool Undefined { [Bind ("isUndefined")] get; }
+
+		[Export ("int64Value")]
+		long Int64Value { get; }
+
+		[Export ("doubleValue")]
+		double DoubleValue { get; }
+
+		[Export ("stringValue")]
+		string StringValue { get; }
+
+		[NullAllowed, Export ("multiArrayValue")]
+		MLMultiArray MultiArrayValue { get; }
+
+		[Export ("dictionaryValue")]
+		NSDictionary<NSObject, NSNumber> DictionaryValue { get; }
+
+#if !WATCH
+		[NullAllowed, Export ("imageBufferValue")]
+		CVPixelBuffer ImageBufferValue { get; }
+
+		[Static]
+		[Export ("featureValueWithPixelBuffer:")]
+		MLFeatureValue FromPixelBuffer (CVPixelBuffer value);
+#endif
+		[Static]
+		[Export ("featureValueWithInt64:")]
+		MLFeatureValue FromInt64 (long value);
+
+		[Static]
+		[Export ("featureValueWithDouble:")]
+		MLFeatureValue FromDouble (double value);
+
+		[Static]
+		[Export ("featureValueWithString:")]
+		MLFeatureValue FromString (string value);
+
+		[Static]
+		[Export ("featureValueWithMultiArray:")]
+		MLFeatureValue FromMultiArray (MLMultiArray value);
+
+		[Static]
+		[Export ("undefinedFeatureValueWithType:")]
+		MLFeatureValue GetUndefinedFeatureValue (MLFeatureType type);
```

Comment:
'Get' sounds wrong since this is still creating and returning an MLFeatureValue same as the other static methods.  FromType doesn't work since it doesn't tell you that it's an UndefinedFeatureValue, but maybe just UndefinedFeatureValueFromType?  or even CreateUndefinedFeatureValue